src/realm/spec.cpp

```diff
@@ -115,50 +99,35 @@ MemRef Spec::create_empty_spec(Allocator& alloc)
     // The 'spec_set' contains the specification (types and names) of
     // all columns and sub-tables
     Array spec_set(alloc);
-    _impl::DeepArrayDestroyGuard dg(&spec_set);
     spec_set.create(Array::type_HasRefs); // Throws
 
-    _impl::DeepArrayRefDestroyGuard dg_2(alloc);
     {
         // One type for each column
         bool context_flag = false;
         MemRef mem = Array::create_empty_array(Array::type_Normal, context_flag, alloc); // Throws
-        dg_2.reset(mem.get_ref());
-        int_fast64_t v(from_ref(mem.get_ref()));
-        spec_set.add(v); // Throws
-        dg_2.release();
+        spec_set.add(from_ref(mem.get_ref()));                                           // Throws
```

Comment:
the old code seems to be very cautious about allocation failure... 
are we safe without it because we can rely on commit rollbacks? Alternatively, could we order things like this? 
```
spec_set.add(0);
MemRef mem = create_empty_array(...);
spec_set.set(spec_set.size() - 1, from_ref(mem.get_ref()));
```

Reply:
I think all these destroy guards are from the time where we could have freestanding tables. They are not needed when we have transactions. Allocations will always be done in the slab area and it will be reset by every commit/rollback (`reset_free_space_tracking()`).

Reply:
Great, thanks for confirming!